app/src/main/java/uk/gov/android/authentication/integrity/keymanager/ECKeyManager.kt

```diff
@@ -43,36 +48,44 @@ class ECKeyManager : KeyStoreManager {
         }
     }
 
-    override fun getPublicKey(): Pair<String, String> {
-        val xByteArr = appCheckPublicKey.w.affineX.toByteArray()
-        val yByteArr = appCheckPublicKey.w.affineY.toByteArray()
-        val x = ProofOfPossessionGenerator.getUrlSafeNoPaddingBase64(xByteArr)
-        val y = ProofOfPossessionGenerator.getUrlSafeNoPaddingBase64(yByteArr)
+    override fun getPublicKeyCoordinates(): Pair<String, String> {
+        val xByteArr = appCheckPublicKey.w.affineX
+        val yByteArr = appCheckPublicKey.w.affineY
+        val xCheckedArray = AppIntegrityUtils
+            .toFixedLengthBytes(xByteArr, EC_POINTS_LENGTH_REQUIREMENT)
+        val yCheckedArray = AppIntegrityUtils
+            .toFixedLengthBytes(yByteArr, EC_POINTS_LENGTH_REQUIREMENT)
+        val x = ProofOfPossessionGenerator.getUrlSafeNoPaddingBase64(xCheckedArray)
+        val y = ProofOfPossessionGenerator.getUrlSafeNoPaddingBase64(yCheckedArray)
         return Pair(x, y)
     }
 
+    override fun getPublicKey(): ECPublicKey {
+        return appCheckPublicKey
+    }
+
     override fun sign(input: ByteArray): ByteArray {
+        val ecSpec = appCheckPublicKey.params
         val signature = Signature.getInstance(ALG).run {
             initSign(appCheckPrivateKeyEntry.privateKey)
             update(input)
             sign()
         }
-        if (BuildConfig.DEBUG) {
-            val verifyResult = verify(input, signature)
-            Log.d("VerifySignature", "$verifyResult")
-        }
-        return signature
+
+        return KeyStoreManager.convertSignatureToASN1(signature, ecSpec)
     }
 
 
-    override fun verify(data: ByteArray, signature: ByteArray): Boolean {
-        val successfulSignature = Signature.getInstance(ALG).run {
-            initVerify(appCheckPrivateKeyEntry.certificate)
-            update(data)
-            verify(signature)
+    override fun verify(jwt: String, jwk: String): Boolean {
+        return try {
+            Jose4jJwtVerifier().verify(jwt, jwk)
```

Comment:
Should this be using the KeyStore verifier, not the Jose4j verifier?

Reply:
We can't do that because the verifier provided from the Keystore is expecting the signature in DER format - and I thought using this ensures the signature is correct and ready to be attached for the JWT - but I can change it and add another function to check it and we can keep both (the KeStore one and the JWTVerifier one)? But the verify method using the KeyStore would then need to be called within the actual `KeyStoreManager.sign()` method before we're converting it. Does that make sense?

Reply:
Understood, I think this is a little tricky and I think (as you've suggested and done) we just remove the verify method for now as we don't actually need the verify method for now.